docs/connect-blockchain/immutablex.mdx

```diff
@@ -0,0 +1,127 @@
+---
+title: Integrate Web3Auth with the ImmutableX Blockchain
+sidebar_label: ImmutableX
+image: "/docs/contents/guides/banners/algorand.png"
+displayed_sidebar: docs
+keywords: [immutablex, web3auth, authentication, blockchain]
+description: "Integrate Web3Auth with the ImmutableX Blockchain | Documentation - Web3Auth"
+---
+
+import GetUserInfoSnippet from "@site/src/common/docs/_get-userinfo.mdx";
+
+---
+
+## Installation
+
+```bash npm2yarn
+npm install --save @imtbl/core-sdk
+```
+
+## Initializing Provider
+
+---
+
+```js
+import { Web3Auth } from "@web3auth/modal";
+import { OpenloginAdapter } from "@web3auth/openlogin-adapter";
+
+const web3auth = new Web3Auth({
+  clientId: "YOUR_WEB3AUTH_CLIENT_ID", // get it from Web3Auth Dashboard
+  chainConfig: {
+    chainNamespace: "eip155", // for all non EVM and SOLANA chains, use "other"
+  },
+});
+
+// "other" is supported through @web3auth/openlogin-adapter package.
+const openloginAdapter = new OpenloginAdapter({
+  adapterSettings: {
+    clientId,
+    network: "testnet",
+    uxMode: "popup",
+  },
+});
+web3auth.configureAdapter(openloginAdapter);
+
+await web3auth.initModal();
+
+const web3authProvider = web3auth.connect(); // web3auth.provider
+```
+
+## Get User Info
+
+---
+
+<GetUserInfoSnippet />
+
+## Creating Layer 1 and Layer 2 Signers
+
+---
+
+Once a user logs in, the Web3Auth SDK returns a provider. ImmutableX is a layer 2 solution for Ethereum, thus we need to create a eth signer using the
+provider. We will use the ethers library to create the signer. Then we use the helper function `createStarkSigner` from the `@imtbl/core-sdk` to
+create a stark signer from the private key.
+
+```js
+import { SafeEventEmitterProvider } from "@web3auth/base";
+import { ImmutableX, Config, createStarkSigner } from "@imtbl/core-sdk";
+import { ethers } from "ethers";
+
+/*
+  Use code from the above Initializing Provider here
+*/
+
+const ethersProvider = new ethers.providers.Web3Provider(web3authProvider);
+const ethSigner = ethersProvider.getSigner();
+
+// Get user's Ethereum public address
+const address = await ethSigner.getAddress();
+
+// Get user's Starkex public address
+
+const starkKey = await web3authProvider.request({ method: "private_key" });
+const starkSigner = createStarkSigner(starkKey);
+```
+
+After creating the signers, we need to register these signers with the ImmutableX SDK. We can do this by calling the `registerOffchain` method.
+
+```js
+const config = Config.SANDBOX; // Or Config.PRODUCTION or Config.ROPSTEN
+const client = new ImmutableX(config);
+
+// We use the signers we created in the above step
+
+const walletConnection = { ethSigner, starkSigner };
+const response = await client.registerOffchain(walletConnection);
+```
+
+## Get Balance
+
+---
+
+We can get the Layer 1 ( Ethereum ) balance using the below code snippet.
```

Comment:
@Saviour1001 Is there a way to get the balance of Layer 2?

Reply:
There are withdrawal methods, but no helper function to get the balance in the immutable core SDK